net/tcp/tcp_send_buffered.c

```diff
@@ -1172,7 +1190,25 @@ ssize_t psock_tcp_send(FAR struct socket *psock, FAR const void *buf,
               goto errout_with_lock;
             }
 
-          net_lockedwait_uninterruptible(&conn->snd_sem);
+          current = TICK2MSEC(clock_systime_ticks());
+          if (expire == UINT_MAX)
+            {
+              remain = UINT_MAX;
+            }
+          else if (expire > current)
+            {
+              remain = expire - current;
+            }
+          else
+            {
+              remain = 0;
+            }
+
+          ret = net_timedwait_uninterruptible(&conn->snd_sem, remain);
```

Comment:
shouldn't this be wrapped with `if (remain > 0)`?

Reply:
which is impossible, 
```

current = TICK2MSEC(clock_systime_ticks());
if (expire == UINT_MAX)
  {
    remain = UINT_MAX;              = UINT_MAX
  }
else if (expire > current)
  {
    remain = expire - current;        > 0
  }
else
  {
    remain = 0;                       = 0
  }

```